Push AV server improvements

fmonniot opened the pull request:
Address some changes introduced in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/11669.

- Switch the webserver from `uvicorn` to `hypercorn`. This unlock HTTP/2 support as well as HTTP/3 (not implemented in this PR though)
  - Note that with hypercorn we are not able to monkey patch TLS client certificate access directly from the code. Instead we need to apply a patch to the library. This is a bit annoying. Unfortunately the TLS client cert specification (on which the patch is based) has been in draft mode for over 3 years without much activity in the various server implementation, so I don't have great hope it will be natively available to us.
  - Another solution would be to use a http proxy (such as nginx) that would do the TLS termination for us. That does complexify the setup quite a bit (more moving pieces) which is the main reason I went with patching the library instead.
- Add support for a `--strict-mode` flag that force CMAF fragments to be uploaded at a specific paths as described in the Matter specification. The non-strict mode is being kept around to help diagnose issues while testing, although we may be able to change that to logging non-accepted paths (it wouldn't be assertable in tests though).
  - This mode is implemented by having each stream have a strict mode attribute. The idea is to be able to mix strict and non strict modes in tests. If we think it's not all that useful, we can also make the mode switch a server's attribute without persistence.

Another thing to note is that the `generate_cmaf_content.sh` script has not yet been updated to follow the latest CMAF Ingest recommendation. That is because I haven't found a way to have ffmpeg push each fragment onto a separate URI. At this point I believe it is not possible, which means we are now looking at something a lot more involved to generate the various media streams.

#### Testing

The tool's README instructions can be followed to test the server itself.

fmonniot committed:
Switch to hypercorn to support HTTP/2 (and HTTP/3 in the future)

fmonniot committed:
Add Hypercorn patch to access TLS info

fmonniot committed:
strict mode for streams

fmonniot committed:
fix file extension & move imports at the top

restyled-commits committed:
Restyled by shfmt

restyled-commits committed:
Restyled by isort

fmonniot committed:
Handle manifest again

fmonniot committed:
Use venv built-in variable

fmonniot committed:
Support for HLS manifest

fmonniot committed:
fix spelling mistakes

fmonniot committed:
Add light interface validation in strict mode

fmonniot committed:
fix typo

fmonniot committed:
Merge remote-tracking branch 'upstream/master' into push-av-server-improvements

fmonniot committed:
Merge remote-tracking branch 'upstream/master' into push-av-server-improvements

fmonniot committed:
Patch hypercorn in chip-cert-bins

chulspro commented on src/tools/push_av_server/requirements.txt:
```diff
@@ -1,5 +1,7 @@
 jinja2
+# We are currently patching hypercorn to add support for the TLS ASGI extension
+# As such updating its version should be linked to an update to the associated patch file
+Hypercorn==0.17.3
```
`ModuleNotFoundError: No module named 'aioquic'` is seen when viewing the push_av_server UI, aioquic module can be included in the requirements. Always appreciated your efforts.

chulspro commented on src/tools/push_av_server/server.py:
```diff
@@ -630,44 +672,40 @@ def __init__(self, host: Optional[str], port: Optional[int], working_directory:
 
         self.app = FastAPI()
         self.app.mount("/static", StaticFiles(directory="static"), name="static")
```
The Static and Templates paths are resolved relative to the push_av_server's directory, this breaks if the server is started outside this directory. Please change this to resolve them relative to the module's directory. Always appreciated your efforts.

fmonniot committed:
Add aioquic to the push av requirements

fmonniot committed:
Use absolute paths for templates/static files

fmonniot committed:
reduce logging for the hpack library

fmonniot committed:
The UI can now correctly display details.json

restyled-commits committed:
Restyled by isort

fmonniot commented:
@chulspro The latest commits should address your comments

fmonniot committed:
Merge branch 'master' into push-av-server-improvements

fmonniot committed:
Merge branch 'master' into push-av-server-improvements

chulspro reviewed (commented):
Thank you for your efforts. please check below result.

fmonniot committed:
Ignore the file names in the patch

And remove the `.patched` extension in the patch file. We are not actually modifying the file name.
